data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/CosmosSqlQueryBuilder.java

```diff
@@ -426,4 +456,28 @@ public Map<String, String> getAdditionalRequiredParameters() {
 
         };
     }
+
+    @NonNull
+    @Override
+    public QueryResult buildPagination(@NonNull Pageable pageable) {
+        int size = pageable.getSize();
+        if (size > 0) {
+            StringBuilder builder = new StringBuilder(" ");
+            long from = pageable.getOffset();
+            builder.append("OFFSET ").append(from).append(" LIMIT ").append(size).append(" ");
+            return QueryResult.of(
+                builder.toString(),
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Collections.emptyMap()
+            );
+        } else {
```

Comment:
Please try to avoid useless `else` when the `if` statement does a return.

Reply:
Will try. Changed https://github.com/micronaut-projects/micronaut-data/pull/1813/commits/7b5265feba176abb69f1b4a78160ac4205b6d1be